crates/loader/src/validation.rs

```diff
@@ -1,19 +1,37 @@
-use anyhow::{anyhow, Result};
-
-const ONLY_ALLOWED_KV_STORE: &str = "default";
+use anyhow::{ensure, Context, Result};
 
 pub(crate) fn validate_key_value_stores(key_value_stores: &Option<Vec<String>>) -> Result<()> {
-    match key_value_stores
-        .iter()
-        .flatten()
-        .find(|id| *id != ONLY_ALLOWED_KV_STORE)
-    {
-        None => Ok(()),
-        Some(invalid) => {
-            let err = anyhow!("Invalid key-value store '{invalid}'. This version of Spin supports only the '{ONLY_ALLOWED_KV_STORE}' store.");
-            Err(err)
-        }
+    for store in key_value_stores.iter().flatten() {
+        validate_component_like_label(store)
+            .with_context(|| format!("Invalid store label {store:?}"))?;
+    }
+    Ok(())
+}
+
+// Validates that the given string is like a component model label, except (currently)
```

Comment:
These rules are very specific (I was especially struck by "it must all all uppercase or all lowercase, no mixing") - is there a particular reason that drives them?  Might be good to comment it if so.  (Or is that what the allusion to the component model is aiming to say?)

Reply:
This attempts to match the `label` rule [from the component model design](https://github.com/WebAssembly/component-model/blob/main/design/mvp/Explainer.md#instance-definitions) to give us forward-compatibility with value import names (once those are implemented).

Reply:
Expanded comment